CMakeLists.txt

```diff
@@ -16,6 +16,14 @@ target_link_libraries(boost_assert
   INTERFACE
     Boost::config
 )
+if (CMAKE_MAJOR_VERSION GREATER_EQUAL 3 AND CMAKE_MINOR_VERSION GREATER_EQUAL 19)
```

Comment:
`if(CMAKE_VERSION VERSION_GREATER 3.18)`

(`VERSION_GREATER_EQUAL` requires CMake 3.7.)

Also, maybe add the CMAKE_GENERATOR check here too, and add the missing empty line before the `if`.

Reply:
> if(CMAKE_VERSION VERSION_GREATER 3.18)

That was the plan, but `VERSION_GREATER` is only available in CMake 3.7 and the project promises support for CMake 3.5.

> Also, maybe add the CMAKE_GENERATOR check here too, and add the missing empty line before the if.

The `CMAKE_GENERATOR` could be one of [Visual Studio 6](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%206.html), [Visual Studio 7](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%207.html), [Visual Studio 7 .NET 2003](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%207%20.NET%202003.html), [Visual Studio 8 2005](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%208%202005.html), [Visual Studio 9 2008](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%209%202008.html), [Visual Studio 10 2010](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2010%202010.html), [Visual Studio 11 2012](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2011%202012.html), [Visual Studio 12 2013](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2012%202013.html), [Visual Studio 14 2015](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2014%202015.html), [Visual Studio 15 2017](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2015%202017.html), [Visual Studio 16 2019](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2016%202019.html), [Visual Studio 17 2022](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2017%202022.html)...

So would you prefer

1)

```cmake
if (CMAKE_GENERATOR MATCHES "Visual Studio.*")
# ...
```

or

2) 

```cmake
string(SUBSTRING CMAKE_GENERATOR 0 13 CMAKE_GENERATOR_SUBSTR)
if (CMAKE_GENERATOR_SUBSTR STREQUAL "Visual Studio")
# ...
```

or 

3) just leave it as it is (I don't think it negatively affects other generators)

?